File: Intranet.Models/CorpComm/History.cs
using System;
using System.ComponentModel;
using System.ComponentModel.DataAnnotations;

namespace Intranet.Models.CorpComm
{
    public class History
    {
        [Key]
        public int Id { get; set; }

        public int CollateralId { get; set; }

        [DisplayName("Request #")]
        public int? RequestId { get; set; }

        [DisplayName("Request Date")]
        public DateTime RequestDate { get; set; } = DateTime.Now;

        [DisplayName("Admin")]
        public string LoginUser { get; set; }

        [DisplayName("Collateral Name")]
        public string CollateralName { get; set; }

        public string EventType { get; set; }
        public int Quantity { get; set; }

        [DisplayName("Station Name / Event Name")]
        public string StationEvent { get; set; }

        public DateTime EventDate { get; set; }
        public DateTime ShippingDate { get; set; }
        public string DropOffPoint { get; set; }
        public int rating { get; set; }
        public string? RejectMarks { get; set; }

        [DisplayName("Adjustment Remarks")]
        public string? ReconRemarks { get; set; }
    }
}

File: Intranet/wwwroot/js/CorpComm/history.js
var dataTable;

$(document).ready(function () {
    loadDataTable();
});

function loadDataTable() {
    dataTable = $('#tblData').DataTable({
        "displayLength": 10,
        "lengthMenu": [[10, 25, 50, -1], [10, 25, 50, "All"]],
        dom: 'Bfrtip',
        "ajax": {
            "url": "/CorpComm/History/GetAll"
        },
        "columns": [
            { "data": "requestId" },
            { "data": "reconRemarks" },
            { "data": "requestDate" },
            { "data": "loginUser" },
            { "data": "collateralName" },
            { "data": "quantity" },
            { "data": "eventType" },
            { "data": "stationEvent" },
            { "data": "eventDate" },
            { "data": "shippingDate" },
            { "data": "dropOffPoint" },
            { "data": "rating" }
        ]
    });
}

function Delete(url) {
    swal({
        title: "Are you sure you want to Delete?",
        text: "You will not be able to restore the data!",
        icon: "warning",
        buttons: true,
        dangerMode: true
    }).then((willDelete) => {
        if (willDelete) {
            $.ajax({
                type: "DELETE",
                url: url,
                success: function (data) {
                    if (data.success) {
                        toastr.success(data.message);
                        dataTable.ajax.reload();
                    }
                    else {
                        toastr.error(data.message);
                    }
                }
            });
        }
    });
}